crates/wasmtime/src/store/func_refs.rs

```diff
@@ -0,0 +1,76 @@
+use crate::module::ModuleRegistry;
+use std::{ptr::NonNull, sync::Arc};
+use wasmtime_runtime::{VMCallerCheckedFuncRef, VMNativeCallHostFuncContext};
+
+/// An arena of `VMCallerCheckedFuncRef`s.
+///
+/// Allows a store to pin and own funcrefs so that it can patch in trampolines
+/// for `VMCallerCheckedFuncRef`s that are missing a `wasm_call` trampoline and
+/// need Wasm to supply it.
+#[derive(Default)]
+pub struct FuncRefs {
+    /// A bump allocation arena where we allocate `VMCallerCheckedFuncRef`s such
+    /// that they are pinned and owned.
+    bump: bumpalo::Bump,
```

Comment:
In retrospect given that this isn't really perf critical or used all that often, perhaps this would be better as a simple `Vec<Box<....>>`?

Reply:
Eh, it is still used whenever passing `funcref`s into Wasm via any method other than function imports that are pre-resolved in `InstancePre`. That is a lot of paths, even if it isn't the hottest path. Since we've already got it, I'm inclined to leave it.